src/back/msl.rs

```diff
@@ -601,6 +579,46 @@ impl<W: Write> Writer<W> {
         }
     }
 
+    fn put_constant<'a>(
+        &mut self,
+        handle: Handle<crate::Constant>,
+        module: &'a crate::Module,
+    ) -> Result<MaybeOwned<'a, crate::TypeInner>, Error> {
+        let constant = &module.constants[handle];
+        let ty = &module.types[constant.ty];
+
+        match &constant.inner {
```

Comment:
please match without a reference here, instead put `ref` in non-movable fields binding, e.g. `constituents`